Neos.Flow/Classes/Http/Client/InternalRequestEngine.php

```diff
@@ -124,10 +126,13 @@ public function sendRequest(ServerRequestInterface $httpRequest): ResponseInterf
         } catch (\Throwable $throwable) {
             $componentContext->replaceHttpResponse($this->prepareErrorResponse($throwable, $componentContext->getHttpResponse()));
         }
-        $session = $this->bootstrap->getObjectManager()->get(SessionInterface::class);
+        $session = $objectManager->get(SessionInterface::class);
         if ($session->isStarted()) {
             $session->close();
         }
+        // FIXME: ObjectManager should forget all instances created during the request
```

Comment:
This case really showed an issue we have with the internal request engine: In case a test is supposed to verify state is correctly rebuilt from an incoming request (e.g. session), the shared ObjectManager currently hides problems.
Probably some other issues are hidden that way and optimally the IRE should completely reset the ObjectManager to a clean state before sending the request.

There is currently no facility for this though, so as a narrow work-around we just forget the SessionManager (to fully get rid of the session so it will be rebuilt from the cookie) and FlashMessageService (to get rid of the cached storage instance, which holds a reference to the FlashMessageContainer).